main.go

```diff
@@ -88,14 +89,21 @@ func (t *tagHandler) getWindow(w http.ResponseWriter, r *http.Request) {
 
 	hour, err := strconv.Atoi(vars["hour"])
 	if err != nil {
-		w.WriteHeader(http.StatusMethodNotAllowed)
+		w.WriteHeader(http.StatusUnprocessableEntity)
 		errMessage := fmt.Sprintf("error parsing hour")
 		_, err := io.WriteString(w, errMessage)
 		fmt.Println(err, errMessage)
 		return
 	}
 
-	tag := t.getTag(day, hour, time.Now())
+	tag, err := t.getTag(day, hour, time.Now())
```

Comment:
Sorry for the unrelated changes. The calculations give an undefined result with invalid params and should thus return an error.